service/src/main/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationServiceImpl.java

```diff
@@ -80,11 +83,47 @@
     @Override
     public PolicyEvaluationResult evalPolicy(final PolicyEvaluationRequestV1 request) {
         ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
-        PolicyEvaluationRequestCacheKey key = new Builder().zoneId(zone.getName()).policySetId("default")
-                .request(request).build();
-        PolicyEvaluationResult result = this.cache.get(key);
+        String uri = request.getResourceIdentifier();
+        String subjectIdentifier = request.getSubjectIdentifier();
+        String action = request.getAction();
+        LinkedHashSet<String> policySetsEvaluationOrder = request.getPolicySetsEvaluationOrder();
+
+        if (uri == null || subjectIdentifier == null || action == null) {
+            LOGGER.error(
+                    String.format(
+                            "Policy evaluation request is missing required input parameters: "
+                                    + "resourceURI='%s' subjectIdentifier='%s' action='%s'",
+                            uri, subjectIdentifier, action));
+
+            throw new IllegalArgumentException("Policy evaluation request is missing required input parameters. "
+                    + "Please review and resubmit the request.");
+        }
+
+        List<PolicySet> allPolicySets = this.policyService.getAllPolicySets();
```

Comment:
Consider transforming this list into a Map of policy names and the respective policies. Later on in the code it seems like we iterate through this list multiple times to search for policies by name. This is O(mn) if we transform the list into a Map we could achieve O(n).